### PR TITLE
[Unit tests] Don't load Poetry cache for Windows

### PR DESCRIPTION
**Pull request by tczekajlo**
**Proposed changes**:
- Use the same key name to restore a cache
- Don't load Poetry cache for Windows
- Include poetry version in a cache key

### CODE REVIEW

**Comment by tczekajlo** [comment]
@m-vdb Can you 👀  and ✅ ? It's a temporary fix for Windows - https://github.com/RasaHQ/rasa/issues/7507

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -228,12 +228,11 @@ jobs:
...
 
     - name: Load Poetry Cached Libraries ⬇
-      if: needs.changes.outputs.backend == 'true'
+      if: needs.changes.outputs.backend == 'true' && matrix.os != 'windows-latest'
       uses: actions/cache@v1
       with:
         path: .venv
-        key: ${{ runner.os }}-poetry-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
-        restore-keys: ${{ runner.os }}-poetry-${{ matrix.python-version }}
+        key: ${{ runner.os }}-poetry-${{ env.POETRY_VERSION }}-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
```
Comment (m-vdb) [inline .github/workflows/continous-integration.yml:235]: do we want to include the `POETRY_VERSION` for the cache key of other workflows?

**Review by m-vdb — APPROVED** [review]
🥥

File: .github/workflows/continous-integration.yml (hunk 1 above)
Comment (tczekajlo) [reply .github/workflows/continous-integration.yml:235]: Hmm, yes! Thanks for pointing it out.

**Pull request merged**